platform/engineering/aws-shell-access.md

```diff
@@ -0,0 +1,79 @@
+# How to Gain Shell Access for Staging/Dev
+## Prerequisites:
+1. [department-of-veterans-affairs/devops](https://github.com/department-of-veterans-affairs/devops) repository checked out
+1. macOS / Win10 + WSL + Ubuntu / Your Favorite *nix
+
+## Actions
+1. Use the [Environment Access Request Template](https://github.com/department-of-veterans-affairs/va.gov-team/issues/new?assignees=&labels=external-request%2C+operations&template=Environment-Access-Request-Template.md&title=Access+for+%5Bindividual%5D) to request AWS console access.
+    * Be sure to follow the instructions within the issue template
+    * You need to have AWS IAM access before you can proceed
+1. Setup your AWS account with MFA and create credentials
+    * Steps borrowed from [aws-account-setup.md](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/engineering/backend/aws-account-setup.md)
+    1. Sign in to AWS web console
+    1. Navigate to IAM dashboard `Users` [here](https://console.amazonaws-us-gov.com/iam/home?region=us-gov-west-1#/home)
+    1. Find your user and click the link on your user name
+    1. Select `Security Credentials` tab
+    1. Locate the `Sign-in credentials` section of the page
+    1. Beside `Assigned MFA device`, click `Manage`
+    1. Install a virtual MFA app on your smarthpone
+        * Google authenticator
+        * Microsoft authenticator
+        * Other options for TOTP codes
+    1. Complete the wizard to pair your virtual MFA
+    1. Locate the `Access keys` section of the page
+    1. Click the `Create access key` button
+        * Use the dialog to download and store the CSV in a secure location
+    1. Use the access key information generated by AWS web to populate a new file `~/.aws/credentials`
+        * Example:
+        ```
+        [default]
+        region = us-gov-west-1
+        0aws_access_key_id = <ACCESS_KEY>
+        aws_secret_access_key = <SECRET_KEY>
+        ```
+1. Setup utility script `issue_mfa.sh`
+    1. Create an alias that includes your IAM username for ease of use:
+        ```bash
+        # create / edit your shell's aliases
+        vi ~/.bash_aliases
+
+        # append to bash_aliases
+        alias mfa="source /path/to/devops/utilities/issue_mfa.sh <your-case-sensitive-IAM-username>"
+        ```
+    1. Reload .bashrc
+        ```bash
+        source ~/.bashrc
+        ```
+    1. Open your virtual MFA app and grab the code
+    1. Test out your alias
+        ```bash
+        mfa <your-mfa-code>
```

Comment:
I ran into a couple of things at this step. It told me what I needed to install and I just followed what it said but it might be worth including as possible steps. They were:

Error: jq is not installed.
Try: brew install jq

Error: aws-cli is not installed.
Try: brew install awscli

Reply:
Nice -- yeah, now that you mention it.  I think I did run into some missing dependencies at this point when I first ran through the steps.  Thanks!